libgnucash/engine/gncCustomer.c

```diff
@@ -358,11 +358,12 @@ static void gncCustomerFree (GncCustomer *cust)
     g_list_free (cust->jobs);
     g_free (cust->balance);
 
-    if (cust->terms)
-        gncBillTermDecRef (cust->terms);
-    if (cust->taxtable)
+    if (!qof_book_shutting_down (qof_instance_get_book (QOF_INSTANCE(cust))))
     {
-        gncTaxTableDecRef (cust->taxtable);
+        if (cust->terms)
+            gncBillTermDecRef (cust->terms);
+        if (cust->taxtable)
+            gncTaxTableDecRef (cust->taxtable);
```

Comment:
This seems wrong at a deeper level: The refcount isn't about memory management, it's about whether a Bill Term or Tax Table is in use by an Owner. The refcount should be decremented only if the Term/Table is disconnected from the Owner or if the Owner is itself deleted by the user.

Reply:
Correct, they are used to to stop the tables that are in use being deleted.
I do not think it is possible to delete a 'Vendor/Customer/Employee'.
I am not at all clear at the moment on the relationship of 'Owner' to the other objects.

Reply:
gncOwner is a struct consisting of a type indicator and a union of gncCustomer, gncEmployee, and gncVendor. That's called a 'discriminated union' and it's similar in concept to the boost::variant we use in KVP and the std::variant in c++options.

So when I say that the Tax Table is in use by an Owner I mean that it's in use by an instance of one of the classes accessible by the Owner union.

The important point is that the TaxTable's and Bill Term's reference count should be incremented when it's added to one of those things and decremented when it's either removed or the thing that references it is deleted from the book, *not* when the owning instance is created/destroyed at session start or end.

Reply:
>  not when the owning instance is created/destroyed at session start or end.

Which, on reinspection, I see that this change does: It prevents the ref counts being decremented at session end.